### ISSUE TITLE
not pulling for addresses with walk scores

### ISSUE THREAD

**Issue by asharma567** [knockrentals/walkscore#3]
Any clue why I can't get a walk score for any of these?

  -- 143 W Fifth Street, Oxnard, CA 93030 34.197966 119.17835
  -- 8520 Hammerly Road, Houston, TX 77055 29.81147 95.49911
  -- 12019 Lake Avenue, Cleveland, OH 44107 41.489484 81.773121
  -- 8400-10100 East Eight Mile Road, Detroit, MI 48234 42.447302 83.019187
  -- 22050 Pennsylvania, Taylor, MI 48180 42.184416 83.248454
  -- 2215 W Southwest Loop 323, Tyler, TX 75701 32.304982 95.328907
  -- 1600 North H Street, Lompoc, CA 93436 34.663782 120.45637
  -- 312 Lafayette Avenue, Hawthorne, NJ 7506 40.948753 74.155605
  -- 322 North Main Avenue, Fallbrook, CA 92028 33.384194 117.251668
  -- 1624 Way St, Reidsville, NC 27320 36.339977 79.659378
  -- 5456 W 34th St, Indianapolis, IN 46224 39.816842 86.255435
  -- 5802-5814 Van Allen Way, Carlsbad, CA 92008 33.136028 117.287423
  -- 1605 King St;, Jacksonville, FL 32204 30.309798 81.692752
  -- 7341 E 6th Ave;, Scottsdale, AZ 85251 33.498726 111.92284
  -- 1215 N Peterson Ave, Douglas, GA 31533 31.522888 82.85102
  -- 2172-2299 Del Carmel Way, Tallahassee, FL 32303 30.473213 84.343172
  -- 3120 E Desert Inn Rd, Las Vegas, NV 89121 36.130363 115.107016
  -- 36 E Washington St, Indianapolis, IN 46204 39.767381 86.156823
  -- 1880 West Oak Parkway; Unit 218, Marietta, GA 30062 33.992297 84.521979
  -- 5220 S. Jones Blvd, Las Vegas, NV 89118 36.093613 115.224436
  -- 3950 River Place Drive, Macon, GA 31021 32.903558 83.68405
  -- 2712 North Central Expressway, Plano, TX 75074 33.034215 96.704091
  -- 1581 Boyles Way, Camp Verde, AZ 86332 34.546465 111.842872
  -- 10267 Beach Drive SW, Calabash, NC 28467 33.889704 78.58216
  -- 194 South Virginia Avenue, Tifton, GA 31794 31.454053 83.529191
  -- 1111 Main Street, St. Helena, CA 94574 38.502829 122.467402
  -- 1740 East Sherman Boulevard, Muskegon, MI 49444 43.17477 86.19917
  -- 1618 Charleston Highway, Columbia, SC 29169 33.969551 81.07829
  -- 100 Poplar Dr, Petersburg, VA 23805 37.214686 77.433233
  -- 775 Elmira Ave, Pasadena, CA 91104 34.159127 118.13396
  -- 4110 Dixie Highway, Louisville, KY 40209 38.188871 85.811303
  -- 175 SW 7th St, Suite 2009, Miami, FL 33130 25.767408 80.197068

**Comment by jayzeng** [comment]
I tried a few of them, I can see walk score return just fine for all of them. Can you post your code snippet?

**Comment by asharma567** [comment]
I know it's not the best design for a persistent connection, just
prototyping stuff as of right now

def get_walk_score_orig(address, lat, lon):
    API_KEY='4f3ee2dbefdd88b9332e12b63b008aa2'
    walkscore = WalkScore(API_KEY)
    input = [address, lat, lon]
    try:
        return walkscore.makeRequest(address, lat, lon)['walkscore']#, input
    except Exception, e:
        print  e, ' --',  ' '.join(map(str,[address, lat, lon]))
        return None

On Thu, Jul 2, 2015 at 3:46 PM, Jay Zeng <notifications@github.com> wrote:

> I tried a few of them, I can see walk score return just fine for all of
> them. Can you post your code snippet?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/knockrentals/walkscore/issues/3#issuecomment-118187017>
> .
>



-- 
Github <https://github.com/asharma567/> | LinkedIn
<http://www.linkedin.com/in/mrasharma/>

**Comment by jayzeng** [comment]
oops, don't paste your api key buddy (I removed it)

**Comment by asharma567** [comment]
Haha thanks! was just going to edit it

**Comment by jayzeng** [comment]
Ok, the problem is these addresses are "being calculated" by walkscore. The code throws an exception (https://github.com/knockrentals/walkscore/blob/master/walkscore/api.py#L72) when status = 2 (see https://www.walkscore.com/professional/api.php), so I guess your best bet is to throw them into a queue, wait for some time and re-query. 

It is pretty clear walk score doesn't do real-time calculation and I think they put them in a queue for batch processing, if I were to guess.

**Comment by jayzeng** [comment]
I do think I should return the raw response from walkscore in the exceptions, what do you think?

**Comment by asharma567** [comment]
yeah that would avoid a lot of debugging headache earlier on, though I'd
make the code flexible s.t. there's an option not to get it back.

On Thu, Jul 2, 2015 at 4:15 PM, Jay Zeng <notifications@github.com> wrote:

> I do think I should return the raw response from walkscore in the
> exceptions, what do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/knockrentals/walkscore/issues/3#issuecomment-118190307>
> .
>



-- 
Github <https://github.com/asharma567/> | LinkedIn
<http://www.linkedin.com/in/mrasharma/>

**Comment by asharma567** [comment]
do you have a quick example of how to queue them once they've failed to an
ScoreUnavailableException , btw you may want to rename that. I got the
impression that there wasn't a score available or maybe separate
jsonRespStatusCode
== 2.
ᐧ

On Thu, Jul 2, 2015 at 4:17 PM, Ajay Sharma <asharma567567@gmail.com> wrote:

> yeah that would avoid a lot of debugging headache earlier on, though I'd
> make the code flexible s.t. there's an option not to get it back.
>
> On Thu, Jul 2, 2015 at 4:15 PM, Jay Zeng <notifications@github.com> wrote:
>
>> I do think I should return the raw response from walkscore in the
>> exceptions, what do you think?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/knockrentals/walkscore/issues/3#issuecomment-118190307>
>> .
>>
>
>
>
> --
> Github <https://github.com/asharma567/> | LinkedIn
> <http://www.linkedin.com/in/mrasharma/>
>



-- 
Github <https://github.com/asharma567/> | LinkedIn
<http://www.linkedin.com/in/mrasharma/>

**Comment by jayzeng** [comment]
Yeah, I am making some changes. Will push soon

**Comment by jayzeng** [comment]
Ok, I will rename ScoreUnavailableException to ScoreBeingCalculatedException, also exception will include the raw response, http status code and a brief description error message.

**Comment by jayzeng** [comment]
Released a new version 0.2.1 to pypi (https://pypi.python.org/pypi/walkscore-api-binding/0.2.1), regarding how to do queue, it will be pretty simple, something like (using sqs as an example)

```
sqs = boto.connect_sqs()
queue_name = 'your_queue'
queue = sqs.get_queue(queue_name)

try: 
  walkscore.makeRequest(add, lat, lng)
except ScoreBeingCalculatedException,e:
  m = Message()
  m.set_body(json.dumps({address: add, lat: lat, lng: lng}))
  queue.write(m)
```

Then you will need a consumer to constantly pull the queue and basically kick off the same logic (ask walk score for score or whatever you have)
